src/node_buffer.cc

```diff
@@ -818,7 +818,7 @@ void IndexOfString(const FunctionCallbackInfo<Value>& args) {
   SPREAD_BUFFER_ARG(args[0], ts_obj);
 
   Local<String> needle = args[1].As<String>();
-  int64_t offset_i64 = args[2]->IntegerValue();
+  int64_t offset_i64 = args[2].As<Integer>()->Value();
```

Comment:
We're not certain that the value is an integer here. The JS code only makes sure it's coerced to a number.